package_manager/util.py

```diff
@@ -1,9 +1,12 @@
 import hashlib
 import base64
 import collections
-import tarfile
 import os
 
+# Distroless's test.sh runs pylint on all python files, but this module will not exist
+# pylint: disable=import-error
+from build_tar import TarFile
```

Comment:
Question as a Python newbie: it's just using a different tar library to create a tar, right?

Reply:
build_tar comes from  "@rules_pkg//:build_tar" . It is the same utility as used by the container layer building tooling.

Internally they also use tarfile as provided by Python, but they add a reproducibility sauce onto that. (such as stripping  user ids by default, resetting mtimes, sanitizing path names...).